src/test/scala/org/kibanaLoadTest/scenario/Dashboard.scala

```diff
@@ -3,132 +3,173 @@ package org.kibanaLoadTest.scenario
             .asJson
             .headers(headers)
             .header("Referer", baseUrl + "/app/dashboards")
             .check(status.is(200))
-        )
-        .exec(
+        ).exec(
+          http("query index pattern meta fields")
+          .get("/api/index_patterns/_fields_for_wildcard")
+            .queryParam("pattern", "kibana_sample_data_ecommerce")
+            .queryParam("meta_fields", "_source")
+            .queryParam("meta_fields", "_id")
+            .queryParam("meta_fields", "_type")
+            .queryParam("meta_fields", "_index")
+            .queryParam("meta_fields", "_score")
+            .headers(headers)
+            .header("Referer", baseUrl + "/app/dashboards")
+            .check(status.is(200))
+        ).exec(
+          http("query index pattern search fields")
+            .get("/api/saved_objects/_find")
+            .queryParam("fields", "title")
+            .queryParam("per_page", "10")
+            .queryParam("search", "kibana_sample_data_ecommerce")
+            .queryParam("search_fields", "title")
+            .queryParam("type", "index-pattern")
+            .headers(headers)
+            .header("Referer", baseUrl + "/app/dashboards")
+            .check(status.is(200))
+        ).exec(
+          http("query input control settings")
+            .get("/api/input_control_vis/settings")
+            .headers(headers)
+            .header("Referer", baseUrl + "/app/dashboards")
+            .check(status.is(200))
+        ).exec(
           http("query timeseries data")
             .post("/api/metrics/vis/data")
-            .body(ElFileBody("data/timeSeriesPayload.json"))
+            .body(StringBody(timeSeriesPayload))
             .asJson
             .headers(headers)
             .header("Referer", baseUrl + "/app/dashboards")
             .check(status.is(200))
-        )
-        .exec(
+        ).exec(
           http("query gauge data")
             .post("/api/metrics/vis/data")
-            .body(ElFileBody("data/gaugePayload.json"))
+            .body(StringBody(gaugePayload))
             .asJson
             .headers(headers)
             .header("Referer", baseUrl + "/app/dashboards")
             .check(status.is(200))
-        )
-    }
+        ).foreach(bSearchPayloadSeq, "index") {
+          exec(session => {
+            session.set(
+              "payloadString",
+              Helper.updateTimeValues(
+                Helper.loadJsonString(s"data/bsearch${session("index").as[Int]}.json"),
+                Map("gte" -> startTime, "lte" -> endTime)
+              )
+            )
+          }).exec(
+            http("query bsearch ${index}")
+              .post("/internal/bsearch")
+              .body(StringBody("${payloadString}"))
+              .asJson
+              .headers(headers)
+              .header("Referer", baseUrl + "/app/dashboards")
+              .check(status.is(200))
+          )
+        }
+      }
```

Comment:
I noticed that number of `bsearch` calls changes time from time. At the moment I'm doing 9 calls with different payload. But what is the right way to replicate it in load testing scenario?

![image](https://user-images.githubusercontent.com/10977896/113872339-15c3e800-97b4-11eb-86f0-6782e2239edb.png)

Reply:
It varies based on how search requests are batched, but ultimately it shouldn't matter how many separate requests you batch it in as long as each individual search request is represented somewhere in one of the `bsearch` requests.